src/components/radio/radio_button.spec.ts

```diff
@@ -0,0 +1,240 @@
+          notificationCount++;
+        });
+      }
+
+      dispatcher.notify('hello');
+
+      expect(notificationCount).toBe(numListeners);
+      done();
+    });
+  });
+
+  describe('MdRadioGroup', () => {
+    let builder: TestComponentBuilder;
+
+    beforeEach(inject([TestComponentBuilder], (tcb: TestComponentBuilder) => {
+      builder = tcb;
+    }));
+
+    it('can select by value', (done: () => void) => {
+      builder
+        .overrideTemplate(TestApp, `
+            <md-radio-group>
+              <md-radio-button value="1"></md-radio-button>
+              <md-radio-button value="2"></md-radio-button>
+            </md-radio-group>`)
+        .createAsync(TestApp)
+        .then((fixture) => {
+          let buttons = getChildDebugElements(fixture.debugElement, 'md-radio-button');
+          let group = getChildDebugElement(fixture.debugElement, 'md-radio-group');
+
+          fixture.detectChanges();
+          expect(group.componentInstance.selected).toBe(null);
+
+          group.componentInstance.value = '2';
+
+          fixture.detectChanges();
+          expect(group.componentInstance.selected).toBe(buttons[1].componentInstance);
+        }).then(done);
+    });
+
+    it('should select uniquely', (done: () => void) => {
+      builder
+        .overrideTemplate(TestApp, `
+            <md-radio-group>
+              <md-radio-button></md-radio-button>
+              <md-radio-button></md-radio-button>
+            </md-radio-group>`)
+        .createAsync(TestApp)
+        .then((fixture) => {
+          let buttons = getChildDebugElements(fixture.debugElement, 'md-radio-button');
+          let group = getChildDebugElement(fixture.debugElement, 'md-radio-group');
+
+          fixture.detectChanges();
+          expect(group.componentInstance.selected).toBe(null);
+
+          group.componentInstance.selected = buttons[0].componentInstance;
+          fixture.detectChanges();
+          expect(isSinglySelected(buttons[0], buttons)).toBe(true);
+
+          group.componentInstance.selected = buttons[1].componentInstance;
+          fixture.detectChanges();
+          expect(isSinglySelected(buttons[1], buttons)).toBe(true);
+        }).then(done);
+    });
+  });
+}
+
+/** Checks whether a given button is uniquely selected from a group of buttons. */
+function isSinglySelected(button: DebugElement, buttons: DebugElement[]): boolean {
+  let component = button.componentInstance;
+  let otherSelectedButtons =
+      buttons.filter((e: DebugElement) =>
+          e.componentInstance != component && e.componentInstance.checked);
+  return component.checked && otherSelectedButtons.length == 0;
+}
+
+/** Gets a child DebugElement by tag name. */
+function getChildDebugElement(parent: DebugElement, selector: string): DebugElement {
+  return parent.query(By.css(selector));
+}
+
+/** Gets an array of child DebugElements by tag name. */
+function getChildDebugElements(parent: DebugElement, selector: string): DebugElement[] {
```

Comment:
You don't need this function any more; sayng `parent.queryAll(By.css(...))` is shorter than calling this function. It was left-over from a time when the `By.css` API didn't exist and you had to give a full predicate function.